plugins/PostProcessingPlugin/scripts/PrintSkewCompensation.py

```diff
@@ -0,0 +1,394 @@
+        if ";POSTPROCESSED" in data[0]:
+            return
+
+        # Notify the user that this script should run first
+        scripts = Application.getInstance().getGlobalContainerStack().getMetaDataEntry("post_processing_scripts")
+        script_list = scripts.split("\\n")
+        if not "PrintSkewCompensation" in script_list[0]:
+            Logger.log("i", str(script_list[0]))
+            Message(
+                title = "[Print Skew Compensation]",
+                text = "Should be first in the Post-Processor list.  It will run if it isn't first, but any following post-processors should act on the changes made by 'Print Skew Compensation'.").show()
+
+        # The current settings for this script
+        self.compensation_method = self.getSettingValueByKey("compensation_method")
+        self.xy_ac_dist = self.getSettingValueByKey("xy_ac_measurement")
+        self.xy_bd_dist = self.getSettingValueByKey("xy_bd_measurement")
+        self.xy_ad_dist = self.getSettingValueByKey("xy_ad_measurement")
+        self.xz_ac_dist = self.getSettingValueByKey("xz_ac_measurement")
+        self.xz_bd_dist = self.getSettingValueByKey("xz_bd_measurement")
+        self.xz_ad_dist = self.getSettingValueByKey("xz_ad_measurement")
+        self.yz_ac_dist = self.getSettingValueByKey("yz_ac_measurement")
+        self.yz_bd_dist = self.getSettingValueByKey("yz_bd_measurement")
+        self.yz_ad_dist = self.getSettingValueByKey("yz_ad_measurement")
+
+        # Skew Factors
+        self.xy_skew_factor = self.calculate_skew_factor(self.xy_ac_dist, self.xy_bd_dist, self.xy_ad_dist)
+        self.xz_skew_factor = self.calculate_skew_factor(self.xz_ac_dist, self.xz_bd_dist, self.xz_ad_dist)
+        self.yz_skew_factor = self.calculate_skew_factor(self.yz_ac_dist, self.yz_bd_dist, self.yz_ad_dist)
+
+        # Select the proper function based on the compensation_method
+        if self.compensation_method == "method_cura":
+            data = self.cura_compensation(data)
+        elif self.compensation_method == "method_marlin":
+            data = self.marlin_compensation(data)
+        elif self.compensation_method == "method_klipper":
+            data = self.klipper_compensation(data)
+
+        # Add the script settings to the gcode if requested
+        if self.getSettingValueByKey("add_settings_to_gcode"):
+            setting_string = f";  Print Skew Compensation Settings:\n;    xy_ac_measurement: {self.xy_ac_dist}\n;    xy_bd_measurement: {self.xy_bd_dist}\n;    xy_ad_measurement: {self.xy_ad_dist}\n;       XY skew factor: {round(self.xy_skew_factor,8)}\n;    xz_ac_measurement: {self.xz_ac_dist}\n;    xz_bd_measurement: {self.xz_bd_dist}\n;    xz_ad_measurement: {self.xz_ad_dist}\n;       XZ skew factor: {round(self.xz_skew_factor,8)}\n;    yz_ac_measurement: {self.yz_ac_dist}\n;    yz_bd_measurement: {self.yz_bd_dist}\n;    yz_ad_measurement: {self.yz_ad_dist}\n;       YZ skew factor: {round(self.yz_skew_factor,8)}\n"
```

Comment:
```suggestion
           setting_string = (
               f";  Print Skew Compensation Settings:\n"
               f";    xy_ac_measurement: {self.xy_ac_dist}\n"
               f";    xy_bd_measurement: {self.xy_bd_dist}\n"
               f";    xy_ad_measurement: {self.xy_ad_dist}\n"
               f";        XY skew factor:    {round(self.xy_skew_factor, 8)}\n"
               f";    xz_ac_measurement: {self.xz_ac_dist}\n"
               f";    xz_bd_measurement: {self.xz_bd_dist}\n"
               f";    xz_ad_measurement: {self.xz_ad_dist}\n"
               f";        XZ skew factor:    {round(self.xz_skew_factor, 8)}\n"
               f";    yz_ac_measurement: {self.yz_ac_dist}\n"
               f";    yz_bd_measurement: {self.yz_bd_dist}\n"
               f";    yz_ad_measurement: {self.yz_ad_dist}\n"
               f";        YZ skew factor:    {round(self.yz_skew_factor, 8)}\n"
           )
```
sorry my initial comment was not clear, this is what I meant with the all at once